### PR TITLE
Desktop fix master

### PR DESCRIPTION
**Pull request by jreidinger**
more then just a merge. Code improved and tests added.

### CODE REVIEW

File: src/lib/installation/clients/desktop_finish.rb [hunk 1]
```diff
@@ -31,230 +31,97 @@
...
+        return nil
+      end
 
-          SCR.Write(path(".sysconfig.windowmanager.DEFAULT_WM"), @default_wm)
-          SCR.Write(
-            path(".sysconfig.windowmanager.X_MOUSE_CURSOR"),
-            @default_cursor
-          )
-          SCR.Write(path(".sysconfig.windowmanager"), nil)
+      desktop_map = DefaultDesktop.GetAllDesktopsMap[selected_desktop]
```
Comment (kobliha) [inline src/lib/installation/clients/desktop_finish.rb]: Maybe raise an exception if there is no desktop_map for a selected_desktop?

File: src/clients/desktop_finish.rb [hunk 2]
```diff
@@ -1,2 +1,2 @@
 require "installation/clients/desktop_finish"
-Yast::DesktopFinishClient.new.main
+Yast::DesktopFinishClient.run
```
Comment (lslezak) [inline src/clients/desktop_finish.rb]: Any reason for this change? I mean I'm fine with a static class method, but we should use the same approach in all clients here. The other client files in the package use the `.new.main` way. It should be consistent...

Comment (jreidinger) [inline src/clients/desktop_finish.rb]: reason is that previously it was simple Yast::Client with common method `main`. Now it is finish client, that have in its API method `run`.

Comment (lslezak) [inline src/clients/desktop_finish.rb]: I see, it's a `FinishClient` now... then OK.

**Pull request merged**